pkg/services/consumer.go

```diff
@@ -63,6 +66,20 @@ func (s *sqlConsumerService) Replace(ctx context.Context, consumer *api.Consumer
 }
 
 func (s *sqlConsumerService) Delete(ctx context.Context, id string) *errors.ServiceError {
+	consumer, err := s.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+	_, e := s.resourceDao.FirstByConsumerName(ctx, consumer.Name)
+	if e == nil {
+		return errors.Forbidden("Resources associated with the consumer: %s", consumer.Name)
+	}
+
+	if e != gorm.ErrRecordNotFound {
+		return handleGetError("Resource", "consumer_name", consumer.Name, e)
+	}
+
+	// e is record not found
```

Comment:
something to consider is, we need to block creation of resource before delete consumer.

Reply:
Thanks @qiujian16! 

I think of two ways to block the resource creation when deleting the consumer.

**Optional 1**: Hard delete the consumer, and leverage the foreign key `fk_resources_consumers` to block creating a resource on it.
  ```bash
  maestro=# \d resources;
                           Table "public.resources"
      Column     |           Type           | Collation | Nullable | Default
  ---------------+--------------------------+-----------+----------+---------
   name          | text                     |           |          |
   id            | text                     |           | not null |
  ...
  Foreign-key constraints:
      "fk_resources_consumers" FOREIGN KEY (consumer_name) REFERENCES consumers(name) ON UPDATE RESTRICT ON DELETE RESTRICT
  ```
  It will throw an error like this to block the creation for the deleted consumer:
  ```bash
  Unable to create Resource: pq: insert or update on table "resources" violates foreign key constraint "fk_resources_consumers"
  ```

**Optional 2**: Keep deleting the consumer softly(mark it as deleted), and add validation to check the consumer when creating resources. If the consumer is not found or marked as deleted, then block the creating option. 
This doesn't seem like an elegant way, as it requires checking each resource creation operation to ensure that.


The operational 1 might be feasible for now. Do you have any other suggestions?